pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper.go

```diff
@@ -142,6 +140,10 @@ func (gls *gitlabScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 	gls.logger.Sugar().Debug("creating a new gitlab client")
 
 	graphClient := graphql.NewClient("https://gitlab.com/api/graphql", gls.client)
+	restClient, err := gitlab.NewClient("", gitlab.WithHTTPClient(gls.client))
```

Comment:
what's the purpose of the `""` empty string? Shouldn't this be the API endpoint and not be empty?

Reply:
That function is looking for a token and a function to be passed in, unfortunately the token is required but is not needed because the authorization is taken care of by the inclusion of the already setup http client.   
<img width="651" alt="Screenshot 2023-09-20 at 8 41 01 AM" src="https://github.com/liatrio/liatrio-otel-collector/assets/42755382/4a2fec61-a4b4-4ab4-90bc-b3456f624231">

Reply:
I think this could be done better. I think we just can use `restClient, err := graphql.NewClient("https://gitlab.com/api/v4", gls.client)` instead. Let's fix that up in a subsequent PR